### DISCUSSION TITLE
Code generation <InvalidType>

### DISCUSSION THREAD

**Discussion by ShyshkovOleg** [rrousselGit/riverpod#2845] [General]
`@riverpod`
`class IngredientsSlice extends _$IngredientsSlice {`
  `  @override`
  `  Stream<RealmResultsChanges<Ingredient>> build() async* {`
    `   final realm = ref.read(realmDbServiceProvider);`
   `    final allIngredients = realm.all<Ingredient>().changes;`
    `   yield* allIngredients;`
  `  }`
`}`

Ingredient - is auto generated class. When I use riverpod code generation inside .g.dart file I get an error: 'Invalid type'

`RealmResultsChanges<InvalidType>>`

Is there any workaround?  Casting types ?

**Comment by rrousselGit** [comment]
This is a known issue. Not much you can do at the moment besides not using a generated type here.

Riverpod 3 should theorically fix this

**Reply by ShyshkovOleg** [reply]
One more question, @rrousselGit.
As I do understand class based provider could not be overridden with .overrideWithValue(...) ?
Error: The method 'overrideWithValue' isn't defined for the type 'NotifierProviderImpl'.
... only with .overrideWith(), but how to use It ?

**Reply by Bilonik** [reply]
Hi @rrousselGit is there any workaround for this? or do you have an ETA for riverpod 3?

Thank you for you awesome work on riverpod.
